modules/aleph/includes/aleph.debt.inc

```diff
@@ -5,8 +5,49 @@
  * Handles debt with Aleph.
  */
 
+use Drupal\aleph\Aleph\Handler\AlephPatronHandler;
+
+/**
+ * Return patron debts.
+ *
+ * @return array
+ *    Array of DingProviderDebt objects containing debts for the patron.
+ */
 function aleph_debt_list() {
-  return array();
+  $result = array();
+  $response = array();
+
+  $patron = aleph_get_patron_by_global_account();
+  $patron_handler = new AlephPatronHandler(aleph_client());
+  $patron_handler->setPatron($patron);
+
+  try {
+    $response = $patron_handler->getDebts();
+  }
+  catch (Exception $e) {
+    watchdog_exception('aleph', $e);
```

Comment:
Without a return here, it continues below, causing an error when you try to call xpath on an empty array.